shared/src/metabase/shared/util.cljc

```diff
@@ -12,3 +15,35 @@
                         (namespace k))]
       (str namespac "/" (name k))
       (name k))))
+
+(defn remove-nils
+  "Given a map, returns a new map with all nil values removed."
+  [m]
+  (m/filter-vals some? m))
+
+(defn normalize-key
+  "Given a string or keyword, in snake_case or kebab-case, normalize it to a :kebab-case keyword.
+
+  Note that namespaces on input keywords are dropped! This is intended for turning idiomatic JSON
+  `\"snake_case_string_keys\"` into idiomatic Clojure `:kebab-case-keywords`."
```

Comment:
Would't it lead to surprises if it handled namespaces?

Reply:
Are you suggesting that I tweak it to preserve the namespaces of any incoming namespaced keys, as a Least Surprise thing? That's a bit more complex logic but I agree it's probably a good idea.

Reply:
Done, take a look.